### DISCUSSION TITLE
how to using Microsoft.Data.SqlClient fluent-nhibernate

### DISCUSSION THREAD

**Discussion by rogeryamato** [nhibernate/fluent-nhibernate#490] [Q&A] [answered]
Is fluent-nhibernate can use Microsoft.Data.SqlClient?

And how to use it..

I see in this issue https://github.com/nhibernate/nhibernate-core/pull/2216 has fix it, But how about for fluent-nhibernate?

Thanks.

**Comment by gliljas** [comment]
Fluent NHibernate is used for mapping and doesn't care how NHibernate connects to the database.

**Comment by jornhd** [comment, marked as answer]
You need to add **.Driver&lt;MicrosoftDataSqlClientDriver&gt;()** in the configuration:

```
Fluently.Configure()
    .Database(
        MsSqlConfiguration.MsSql2012
            .Driver<MicrosoftDataSqlClientDriver>()
            .ConnectionString(connectionString)
    )
    .Mappings(...)
    .BuildSessionFactory();
```

**Reply by storm-caller** [reply]
Hi, sorry. I do not quie get it. Is it real class "MicrosoftDataSqlClientDriver" or something that I need implement/derive mysel? using Microsoft.Data.SqlClient; – no such class there.

**Reply by hazzik** [reply]
It is in NHibernate

**Reply by hubocan** [reply]
It has been introduced in version NHibernate 5.3.0, so make sure you use at least this version.
